sympy/polys/matrices/domainmatrix.py

```diff
@@ -3256,6 +3256,23 @@ def lu(self):
         L, U, swaps = self.rep.lu()
         return self.from_rep(L), self.from_rep(U), swaps
 
+    def qr(self):
+        """
+        QR decomposition for DomainMatrix.
+
+        Returns:
+            - Q: Orthogonal DomainMatrix.
+            - R: Upper triangular DomainMatrix.
+        """
+        if isinstance(self.rep, DFM):
+            ddm_q, ddm_r = self.rep.qr()
+        else:
+            ddm_q, ddm_r = self.rep.qr()
```

Comment:
If the code is the same in both branches then the if check is not needed.

Reply:
Okay, I will update that.